### PR TITLE
[BEAM-13031] [Playground] Code editor - Support Localization

### PR DESCRIPTION
**Pull request by miamihotline**
Added localization functionality, localization instructions to README.md
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

`ValidatesRunner` compliance status (on master branch)
--------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Lang</th>
      <th>ULR</th>
      <th>Dataflow</th>
      <th>Flink</th>
      <th>Samza</th>
      <th>Spark</th>
      <th>Twister2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Go</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/">
          Build Status
        </a>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
    <tr>
      <td>Java</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_ULR/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Streaming/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2_Streaming/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
    <tr>
      <td>Python</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
    <tr>
      <td>XLang</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
  </tbody>
</table>

Examples testing status on various runners
--------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Lang</th>
      <th>ULR</th>
      <th>Dataflow</th>
      <th>Flink</th>
      <th>Samza</th>
      <th>Spark</th>
      <th>Twister2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Go</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>Java</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Java11_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_Examples_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>Python</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>XLang</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
  </tbody>
</table>

Post-Commit SDK/Transform Integration Tests Status (on master branch)
------------------------------------------------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Go</th>
      <th>Java</th>
      <th>Python</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
  </tbody>
</table>

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>---</th>
      <th>Java</th>
      <th>Python</th>
      <th>Go</th>
      <th>Website</th>
      <th>Whitespace</th>
      <th>Typescript</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Non-portable</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
    <tr>
      <td>Portable</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_GoPortable_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
  </tbody>
</table>

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Review by damondouglas — CHANGES_REQUESTED** [review]
Hellow @miamihotline when I run `./gradlew playgroundPrecommit`, I receive the following output:

```
> Task :playground:frontend:pubGet
Running "flutter pub get" in frontend...                         1,766ms

> Task :playground:frontend:analyze
Analyzing ....

  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:49:12 - The method 'getPrecompiledObjects' isn't defined for the type 'PlaygroundServiceClient'. Try correcting the name to the name of an existing method, or defining a method named 'getPrecompiledObjects'. - undefined_method
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:60:12 - The method 'getPrecompiledObjectCode' isn't defined for the type 'PlaygroundServiceClient'. Try correcting the name to the name of an existing method, or defining a method named 'getPrecompiledObjectCode'. - undefined_method
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:69:12 - The method 'getPrecompiledObjectOutput' isn't defined for the type 'PlaygroundServiceClient'. Try correcting the name to the name of an existing method, or defining a method named 'getPrecompiledObjectOutput'. - undefined_method
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:82:3 - Undefined class 'grpc.GetPrecompiledObjectsRequest'. Try changing the name to the name of an existing class, or creating a class with the name 'grpc.GetPrecompiledObjectsRequest'. - undefined_class
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:85:17 - The function 'GetPrecompiledObjectsRequest' isn't defined. Try importing the library that defines 'GetPrecompiledObjectsRequest', correcting the name to the name of an existing function, or defining a function named 'GetPrecompiledObjectsRequest'. - undefined_function
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:92:3 - Undefined class 'grpc.GetPrecompiledObjectRequest'. Try changing the name to the name of an existing class, or creating a class with the name 'grpc.GetPrecompiledObjectRequest'. - undefined_class
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:95:17 - The function 'GetPrecompiledObjectRequest' isn't defined. Try importing the library that defines 'GetPrecompiledObjectRequest', correcting the name to the name of an existing function, or defining a function named 'GetPrecompiledObjectRequest'. - undefined_function
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:126:38 - Undefined class 'grpc.PrecompiledObjectType'. Try changing the name to the name of an existing class, or creating a class with the name 'grpc.PrecompiledObjectType'. - undefined_class
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:128:12 - Case expressions must be constant. - non_constant_case_expression
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:128:12 - Case expressions must be constant. - non_constant_case_expression
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:128:17 - The name 'PrecompiledObjectType' is being referenced through the prefix 'grpc', but it isn't defined in any of the libraries imported using that prefix. Try correcting the prefix or importing the library that defines 'PrecompiledObjectType'. - undefined_prefixed_name
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:130:12 - Case expressions must be constant. - non_constant_case_expression
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:130:12 - Case expressions must be constant. - non_constant_case_expression
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:130:17 - The name 'PrecompiledObjectType' is being referenced through the prefix 'grpc', but it isn't defined in any of the libraries imported using that prefix. Try correcting the prefix or importing the library that defines 'PrecompiledObjectType'. - undefined_prefixed_name
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:132:12 - Case expressions must be constant. - non_constant_case_expression
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:132:12 - Case expressions must be constant. - non_constant_case_expression
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:132:17 - The name 'PrecompiledObjectType' is being referenced through the prefix 'grpc', but it isn't defined in any of the libraries imported using that prefix. Try correcting the prefix or importing the library that defines 'PrecompiledObjectType'. - undefined_prefixed_name
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:148:48 - The getter 'precompiledObjects' isn't defined for the type 'Categories_Category'. Try importing the library that defines 'precompiledObjects', correcting the name to the name of an existing getter, or defining a getter or field named 'precompiledObjects'. - undefined_getter
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:149:37 - 3 positional argument(s) expected, but 0 found. Try adding the missing arguments. - not_enough_positional_arguments
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:150:19 - The named parameter 'name' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'name'. - undefined_named_parameter
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:151:19 - The named parameter 'description' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'description'. - undefined_named_parameter
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:152:19 - The named parameter 'type' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'type'. - undefined_named_parameter
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:153:19 - The named parameter 'path' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'path'. - undefined_named_parameter
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:156:43 - 2 positional argument(s) expected, but 0 found. Try adding the missing arguments. - not_enough_positional_arguments
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:157:11 - The named parameter 'name' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'name'. - undefined_named_parameter
  error - build/lib/modules/examples/repositories/example_client/grpc_example_client.dart:158:11 - The named parameter 'examples' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'examples'. - undefined_named_parameter
  error - build/lib/pages/playground/states/example_selector_state.dart:37:32 - The default value of an optional parameter must be constant. - non_constant_default_value
  error - build/lib/pages/playground/states/example_selector_state.dart:37:44 - There's no constant named 'all' in 'ExampleType'. Try correcting the name to the name of an existing constant, or defining a constant named 'all'. - undefined_enum_constant
  error - build/lib/pages/playground/states/example_selector_state.dart:61:38 - The method 'getCategories' isn't defined for the type 'ExampleState'. Try correcting the name to the name of an existing method, or defining a method named 'getCategories'. - undefined_method
  error - build/lib/pages/playground/states/example_selector_state.dart:63:41 - 2 positional argument(s) expected, but 0 found. Try adding the missing arguments. - not_enough_positional_arguments
  error - build/lib/pages/playground/states/example_selector_state.dart:64:13 - The named parameter 'name' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'name'. - undefined_named_parameter
  error - build/lib/pages/playground/states/example_selector_state.dart:65:13 - The named parameter 'examples' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'examples'. - undefined_named_parameter
  error - build/lib/pages/playground/states/example_selector_state.dart:72:63 - There's no constant named 'all' in 'ExampleType'. Try correcting the name to the name of an existing constant, or defining a constant named 'all'. - undefined_enum_constant
  error - build/test/pages/playground/states/example_selector_state_test.dart:39:57 - Too many positional arguments: 0 expected, but 1 found. Try removing the extra arguments. - extra_positional_arguments
  error - build/test/pages/playground/states/example_selector_state_test.dart:50:61 - Too many positional arguments: 0 expected, but 1 found. Try removing the extra arguments. - extra_positional_arguments
  error - build/test/pages/playground/states/example_selector_state_test.dart:63:57 - Too many positional arguments: 0 expected, but 1 found. Try removing the extra arguments. - extra_positional_arguments
  error - build/test/pages/playground/states/example_selector_state_test.dart:71:27 - The getter 'sdkCategories' isn't defined for the type 'ExampleState'. Try importing the library that defines 'sdkCategories', correcting the name to the name of an existing getter, or defining a getter or field named 'sdkCategories'. - undefined_getter
  error - build/test/pages/playground/states/example_selector_state_test.dart:71:55 - The getter 'sdkCategories' isn't defined for the type 'ExampleState'. Try importing the library that defines 'sdkCategories', correcting the name to the name of an existing getter, or defining a getter or field named 'sdkCategories'. - undefined_getter
  error - build/test/pages/playground/states/example_selector_state_test.dart:82:57 - Too many positional arguments: 0 expected, but 1 found. Try removing the extra arguments. - extra_positional_arguments
  error - build/test/pages/playground/states/example_selector_state_test.dart:90:27 - The getter 'sdkCategories' isn't defined for the type 'ExampleState'. Try importing the library that defines 'sdkCategories', correcting the name to the name of an existing getter, or defining a getter or field named 'sdkCategories'. - undefined_getter
  error - build/test/pages/playground/states/example_selector_state_test.dart:90:55 - The getter 'sdkCategories' isn't defined for the type 'ExampleState'. Try importing the library that defines 'sdkCategories', correcting the name to the name of an existing getter, or defining a getter or field named 'sdkCategories'. - undefined_getter
  error - build/test/pages/playground/states/example_selector_state_test.dart:103:57 - Too many positional arguments: 0 expected, but 1 found. Try removing the extra arguments. - extra_positional_arguments
  error - build/test/pages/playground/states/example_selector_state_test.dart:111:27 - The getter 'sdkCategories' isn't defined for the type 'ExampleState'. Try importing the library that defines 'sdkCategories', correcting the name to the name of an existing getter, or defining a getter or field named 'sdkCategories'. - undefined_getter
  error - build/test/pages/playground/states/example_selector_state_test.dart:111:55 - The getter 'sdkCategories' isn't defined for the type 'ExampleState'. Try importing the library that defines 'sdkCategories', correcting the name to the name of an existing getter, or defining a getter or field named 'sdkCategories'. - undefined_getter
  error - build/test/pages/playground/states/mocks/categories_mock.dart:25:16 - 2 positional argument(s) expected, but 0 found. Try adding the missing arguments. - not_enough_positional_arguments
  error - build/test/pages/playground/states/mocks/categories_mock.dart:25:17 - The named parameter 'name' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'name'. - undefined_named_parameter
  error - build/test/pages/playground/states/mocks/categories_mock.dart:25:33 - The named parameter 'examples' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'examples'. - undefined_named_parameter
  error - build/test/pages/playground/states/mocks/categories_mock.dart:25:44 - Undefined name 'exampleMock1'. Try correcting the name to one that is defined, or defining the name. - undefined_identifier
  error - build/test/pages/playground/states/mocks/categories_mock.dart:26:16 - 2 positional argument(s) expected, but 0 found. Try adding the missing arguments. - not_enough_positional_arguments
  error - build/test/pages/playground/states/mocks/categories_mock.dart:26:17 - The named parameter 'name' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'name'. - undefined_named_parameter
  error - build/test/pages/playground/states/mocks/categories_mock.dart:26:35 - The named parameter 'examples' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'examples'. - undefined_named_parameter
  error - build/test/pages/playground/states/mocks/categories_mock.dart:26:46 - Undefined name 'exampleMock2'. Try correcting the name to one that is defined, or defining the name. - undefined_identifier
  error - build/test/pages/playground/states/mocks/categories_mock.dart:30:16 - 2 positional argument(s) expected, but 0 found. Try adding the missing arguments. - not_enough_positional_arguments
  error - build/test/pages/playground/states/mocks/categories_mock.dart:30:17 - The named parameter 'name' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'name'. - undefined_named_parameter
  error - build/test/pages/playground/states/mocks/categories_mock.dart:30:33 - The named parameter 'examples' isn't defined. Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'examples'. - undefined_named_parameter
  error - build/test/pages/playground/states/mocks/categories_mock.dart:30:44 - Undefined name 'exampleMock1'. Try correcting the name to one that is defined, or defining the name. - undefined_identifier
  error - build/test/pages/playground/states/mocks/categories_mock.dart:33:46 - Undefined name 'exampleMock1'. Try correcting the name to one that is defined, or defining the name. - undefined_identifier
  error - build/test/pages/playground/states/mocks/categories_mock.dart:33:60 - Undefined name 'exampleMock2'. Try correcting the name to one that is defined, or defining the name. - undefined_identifier
  error - build/test/pages/playground/states/mocks/categories_mock.dart:35:54 - Undefined name 'exampleMock2'. Try correcting the name to one that is defined, or defining the name. - undefined_identifier
  error - build/test/pages/playground/states/mocks/categories_mock.dart:37:54 - Undefined name 'exampleMock1'. Try correcting the name to one that is defined, or defining the name. - undefined_identifier
   info - build/test/pages/playground/states/mocks/categories_mock.dart:22:8 - Unused import: 'example_mock.dart'. Try removing the import directive. - unused_import
   info - lib/modules/actions/components/reset_action.dart:28:26 - Only use double quotes for strings containing single quotes. - prefer_single_quotes
   info - lib/modules/output/components/output_area.dart:21:8 - Unused import: 'package:playground/constants/sizes.dart'. Try removing the import directive. - unused_import
   info - lib/pages/playground/playground_page.dart:32:8 - Duplicate import. Try removing all but one import of the library. - duplicate_import
   info - lib/pages/playground/playground_page.dart:33:8 - Duplicate import. Try removing all but one import of the library. - duplicate_import
   info - lib/pages/playground/playground_page.dart:34:8 - Duplicate import. Try removing all but one import of the library. - duplicate_import
   info - lib/pages/playground/playground_page.dart:35:8 - Duplicate import. Try removing all but one import of the library. - duplicate_import
   info - lib/pages/playground/playground_page.dart:36:8 - Duplicate import. Try removing all but one import of the library. - duplicate_import
   info - lib/pages/playground/playground_page.dart:37:8 - Duplicate import. Try removing all but one import of the library. - duplicate_import
   info - lib/pages/playground/playground_page.dart:38:8 - Duplicate import. Try removing all but one import of the library. - duplicate_import

70 issues found.

> Task :playground:frontend:analyze FAILED
```

**Comment by miamihotline** [comment]
> Hellow @miamihotline when I run `./gradlew playgroundPrecommit`, I receive the following output:

Hello, @damondouglas  I don't see such issues, anyways I fixed the ones that were in this PR.
image

**Comment by aaltay** [comment]
What is the next step on this PR?

File: playground/frontend/lib/l10n/app_en.arb [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{
```
Comment (damondouglas) [inline playground/frontend/lib/l10n/app_en.arb:1]: This file may be what is causing the rat task failure.  You can run the rat task on your machine:
```
cd /path/to/apache/beam/on/your/machine
./gradlew rat
```

**Review by damondouglas — CHANGES_REQUESTED** [review]

File: playground/frontend/lib/l10n/app_en.arb (hunk 1 above)
Comment (miamihotline) [reply playground/frontend/lib/l10n/app_en.arb:1]: Hello @damondouglas, thank you for reviewing , yes, i think i forgot to add *.arb files to analysis exclusions, i'll fix this issue in 5 minutes

**Comment by miamihotline** [comment]
R:@pabloem

**Review by damondouglas — APPROVED** [review]
@pabloem LGTM

**Pull request merged**